### PR TITLE
Shallow fusion for Aishell

### PR DESCRIPTION
**Pull request by marcoyang1998**
This PR adds the results of decoding with RNNLM shallow fusion and LODR for Aishell.
The RNNLM is trained with Aishell1 and Aishell2 text (many thanks to #945). The pre-trained RNNLM can be found here https://huggingface.co/marcoyang/icefall-aishell-rnn-lm. 

For the decoding results, please refer to RESULTS.md.

### CODE REVIEW

**Review by yfyeung — APPROVED** [review]

**Comment by marcoyang1998** [comment]
> The table of results is;
> 
> decoding method	test	dev	
> greedy search	5.39	5.09	--epoch 29 --avg 5 --max-duration 600
> modified beam search	5.05	4.79	--epoch 29 --avg 5 --max-duration 600
> modified beam search + RNNLM shallow fusion	3.08	2.81	--epoch 29 --avg 5 --max-duration 600
> modified beam search + LODR	2.72	2.51	--epoch 29 --avg 5 --max-duration 600
> fast beam search	5.13	4.91	--epoch 29 --avg 5 --max-duration 600
> .. why is there so much improvement from LM rescoring in this scenario?

It seems that part of the validation set is included in the LM training set. I will fix this and update the results.

**Comment by marcoyang1998** [comment]
An update to the LM fusion result.

The previous large improvements after LM fusion were caused by an overlap between the Aishell1 dev/test set and Aishell2 train set. After fixing this, the relative WER reduction is much smaller.

| decoding_method | dev | test |
| -------------------- | ---- | ---- |
| modified_beam_search | 4.79  |  5.05  |
| RNNLM shallow fusion   | 4.53 | 4.73 |
| LODR | 4.37 | 4.57 |
